File: Day8/main.ts
import fs from "fs";

const data = fs.readFileSync("./input.txt", "utf-8").split("\n");

const seq = data[0];
const raw_network = data.slice(2);
const network = {};

for (const node of raw_network) {
  network[node.split(" = ")[0]] = node
    .split(" = ")[1]
    .trim()
    .slice(1, -1)
    .split(", ");
}

let count = 0;
let node = "AAA";
const LRtoIdx = (x: string) => (x === "L" ? 0 : 1);

while (true) {
  const new_node = network[node][LRtoIdx(seq[count % seq.length])];
  node = new_node;
  count += 1;
  if (new_node === "ZZZ") break;
}

console.log(count);

File: Day8/main.py
with open("./Day8/input.txt") as f:
    data = f.readlines()

seq = data[0]
raw_network = data[2:]
network = {node.split(" = ")[0]: node.split(
    " = ")[1].strip()[1:-1].split(", ") for node in raw_network}

count = 0
node = "AAA"


def LRtoIdx(x): return 0 if x == "L" else 1


while True:
    new_node = network[node][LRtoIdx(seq[count % len(seq)])]
    count += 1
    node = new_node
    if new_node == "ZZZ":
        break

print(count)
